src/board.js

```diff
@@ -483,3 +488,72 @@ function clamp(value, min, max) {
   }
   return value;
 }
+
+class PinUI {
+  constructor(element, label) {
+    this.label = label;
+    this._isTouched = false;
+
+    this.element = element;
+    this.element.setAttribute("role", "button");
+    this.element.setAttribute("tabindex", "0");
+    this.element.ariaLabel = label;
+    this.element.style.cursor = "pointer";
+
+    this.keyListener = (e) => {
+      switch (e.key) {
+        case "Enter":
+        case " ":
+          e.preventDefault();
+          if (e.type === "keydown") {
+            this.press();
+          } else {
+            this.release();
+          }
+      }
+    };
+
+    this.mouseDownListener = (e) => {
+      e.preventDefault();
+      this.press();
+    };
+    this.mouseUpListener = (e) => {
+      e.preventDefault();
+      this.release();
+    };
+    this.mouseLeaveListener = (e) => {
+      this.release();
+    };
+
+    this.element.addEventListener("mousedown", this.mouseDownListener);
+    this.element.addEventListener("mouseup", this.mouseUpListener);
+    this.element.addEventListener("keydown", this.keyListener);
+    this.element.addEventListener("keyup", this.keyListener);
+    this.element.addEventListener("mouseleave", this.mouseLeaveListener);
+  }
+
+  press() {
+    this._isTouched = true;
+    this.render();
+  }
+
+  release() {
+    this._isTouched = false;
+    this.render();
+  }
+
+  isTouched() {
+    return this._isTouched;
+  }
+
+  render() {
+    const fill = this._isTouched ? "red" : "url(#an)";
```

Comment:
More appropriate colour than red?

Reply:
I think we can get this in and discuss the UI of the buttons/logo later. The colour choice for the buttons was similarly arbitrary. MakeCode fill the background of the logo which is a bit misleading but may well have been discussed at length.